code/game/atoms.dm

```diff
@@ -362,40 +362,125 @@
 		A.fingerprintshidden |= fingerprintshidden.Copy()    //admin
 	A.fingerprintslast = fingerprintslast
 
+var/list/blood_splatter_icons = list()
 
-//returns 1 if made bloody, returns 0 otherwise
-/atom/proc/add_blood(mob/living/carbon/human/M as mob)
+/atom/proc/blood_splatter_index()
+	return "\ref[initial(icon)]-[initial(icon_state)]"
 
-	if(!blood_DNA || !istype(blood_DNA, /list))	//if our list of DNA doesn't exist yet (or isn't a list) initialise it.
-		blood_DNA = list()
+//returns the mob's dna info as a list, to be inserted in an object's blood_DNA list
+/mob/living/proc/get_blood_dna_list()
+	if(get_blood_id() != "blood")
+		return
+	return list("ANIMAL DNA" = "Y-")
 
-	blood_color = "#A10808"
-	if(istype(M))
-		if(M.species.flags & NO_BLOOD)
-			return 0
-		M.check_dna()
-		blood_color = M.species.blood_color
+/mob/living/carbon/get_blood_dna_list()
+	if(get_blood_id() != "blood")
+		return
+	var/list/blood_dna = list()
+	if(dna)
+		var/mob/living/carbon/human/H = src
+		blood_dna[dna.unique_enzymes] = H.b_type
+	else
+		blood_dna["UNKNOWN DNA"] = "X*"
+	return blood_dna
 
-	. = 1
-	return 1
+/mob/living/carbon/alien/get_blood_dna_list()
+	return list("UNKNOWN DNA" = "X*")
 
-/atom/proc/add_blood_list(mob/living/carbon/M)
-	// Returns 0 if we have that blood already
-	if(!istype(blood_DNA, /list))	//if our list of DNA doesn't exist yet (or isn't a list) initialise it.
+//to add a mob's dna info into an object's blood_DNA list.
+/atom/proc/transfer_mob_blood_dna(mob/living/L)

```

Comment:
Yeah if I recall our dna system vs there's and our use of how blood coloring is stored and accessed